### ISSUE TITLE
Speed accuracy estimate - e.g. SDOP / SDOS / EHVE / sAcc equivalent

### ISSUE THREAD

**Issue by Logiqx** [Logiqx/gp3s-coros#8]
TLDR - It appears this is not possible with the Sony chip. See the comment on 28/11/2022.

It is not clear whether the equivalent is available from the Sony chip but in speedsurfing we make use of SDOP / SDOS (Locosys) and sAcc (u-blox).

SDOP was introduced in the SiRFstar III and was first made available to GP3S in the Locosys GT-31. It was also available in data from the Locosys GW-52 and GW-60 which probably use the SiRFstar IV.

The usefulness of SDOP was well documented by Tom Chalko in his [paper from 2009](https://nujournal.net/estimating-accuracy-of-gps-doppler-speed-measurement-using-speed-dilution-of-precision-sdop-parameter/).

SDOP is calculated by the GPS chip and is based using the covariation matrix within the Kalman filter. It is essentially giving a +/- figure for the uncertainty in the speed, based on 3 sigma (99.7%).

sAcc is similar to SDOP / SDOS (in principle) and is available from u-blox GPS chips. It is therefore available from GPS devices such as the Motion, Gyro1 and ESP-GPS which all use u-blox 8 variants. It is more than likely calculated in the same way as SDOP (using the covariance matrix) but is based on 1 standard deviation.

A recent [investigation](https://logiqx.github.io/gps-guides/detective.html) that I completed suggest that SDOP / SDOS may be derived in a similar way to the SiRF EHVE parameter but the important point is that it can only be done within the GPS / GNSS chip itself as it makes use of data within the Kalman filter.

It is unclear whether an SDOP / SDOS / EHVE / sAcc equivalent exists for the Sony chip (APEX Pro / VERTIX) or Airoha AG3335M (VERTIX 2). I have yet to see any such item in a MediaTek or Airoha specification. I can find no reference for any NMEA commands relating to this kind of data but I have no access to the technical specification for the Sony or Airoha chips used by COROS.

If Sony or Airoha do provide speed accuracy estimates then please add them to the FIT data so that it can be used for subsequent analysis of results.

If this data is unavailable then it would be useful if this point can be confirmed.

**Comment by Logiqx** [comment]
I have stumbled across the user's manual for the Sony CXD5603GF GNSS chip which I believe is in the APEX Pro and VERTIX:

https://github.com/Xinyuan-LilyGO/LilyGO-T-Motion/blob/master/datasheet/CXD5603GF_UsersManual_E_013_TianQin.pdf

It is clear that the Sony chip cannot provide speed accuracy data as it only outputs the 8 standard NMEA sentences; GGA, GLL, GNS, GSA, GSV, RMV, VTG, ZDA. I am very familiar with these NMEA sentences and I know that speed accuracy is not included.

I have applied the label "wontfix" but I have left the issue open so that this recent finding is clearly visible.

**Comment by Logiqx** [comment]
Closing this issue because it cannot be implemented.
